sw/device/lib/crypto/impl/hash.c

```diff
@@ -211,127 +212,91 @@ crypto_status_t otcrypto_hash(crypto_const_byte_buf_t input_message,
     return OTCRYPTO_BAD_ARGS;
   }
 
-  // Check `digest->len` is consistent with `hash_mode`
-  HARDENED_TRY(check_digest_len(hash_mode, digest->len));
+  // Check that digest length and mode match.
+  HARDENED_TRY(check_digest_len(digest));
 
-  switch (hash_mode) {
+  switch (digest->mode) {
```

Comment:
nit: Should we consider adding a `HARDENED_CHECK_EQ` macro to make the `digest->mode` check redundant inside this function? Probably low priority depending on how the enum is encoded.

Reply:
Yes, there are a lot of `HARDENED_CHECK_EQ`s I want to add to these files under `impl`. But to do it properly I need to compare to the disassembly and add `launder32` if the compiler is optimizing away the check, so I planned to add these extra hardening checks in a batch as a separate pass over the cryptolib functions.

Reply:
SG. Thanks!